azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/CallHistoryService.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.service
+
+import com.azure.android.communication.ui.calling.data.CallHistoryRepository
+import com.azure.android.communication.ui.calling.redux.Store
+import com.azure.android.communication.ui.calling.redux.state.ReduxState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import org.threeten.bp.LocalDateTime
+
+internal interface CallHistoryService {
+    fun start(coroutineScope: CoroutineScope)
+}
+
+internal class CallHistoryServiceImpl(
+    private val store: Store<ReduxState>,
+    private val callHistoryRepository: CallHistoryRepository,
+) : CallHistoryService {
+    private val callIdStateFlow = MutableStateFlow<String?>(null)
+
+    override fun start(coroutineScope: CoroutineScope) {
```

Comment:
Since this will actually do an `insert`, I would call this something like `recordCallStart` so that it's obvious to the callers what happens. Otherwise this `insert` almost looks like a non-obvious side-effect of "starting a service".

Also there's no corresponding "stop" method for this service, so I'd definitely not using the name "start" for this.

Reply:
As per our conversation this method does not do insert right away, it will launch observing on the redux state change.